src/App.tsx

```diff
@@ -1,61 +1,124 @@
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Post } from './types/post';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUserById(userId: number) {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+const todos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+function getNewPostId(posts: Post[]) {
+  const maxId = Math.max(...posts.map(post => post.id));
+
+  return maxId + 1;
+}
+
+export const App: React.FC = () => {
+  const [posts, setPosts] = useState(todos);
+
+  function addTodosInArr({ id, ...data }: Post) {
+    const newPost = {
+      id: id || getNewPostId(posts),
+      ...data,
+    };
+
+    setPosts(currentPosts => [...currentPosts, newPost]);
+  }
+
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+
+    setHasTitleError(false);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
```

Comment:
[CODE STYLE] - Don't use one-letter variable naming. It could be confusing for another developer, who reads your code. Also, it would really hard to find this variable in the code editor using search. Consider renaming `e` to `event` or `changeEvent`.